intel-sgx/fortanix-sgx-tools/src/bin/ftxsgx-runner.rs

```diff
@@ -47,6 +48,30 @@ fn catch_sigbus() {
     }
 }
 
+#[cfg(unix)]
+fn catch_sigusr1() {
+    unsafe {
+        extern "C" fn handle_sigusr1(_signo: c_int, _info: *mut siginfo_t, _context: *mut c_void) {
+            eprintln!("SIGUSR1 triggered, thread_id: {:?}", std::thread::current().id());
+            let instruction_ptr = unsafe { (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] as *const u8};
+            const ENCLU: [u8; 3] = [0x0f, 0x01, 0xd7];
+            let is_enclu = ENCLU.iter().enumerate().all(|(idx, v)| {
+                unsafe { *instruction_ptr.offset(idx as isize) == *v }
+            });
+            if is_enclu {
+                // At enclu instruction - force IP to the next instruction after enclu
+                unsafe { (*(_context as *mut ucontext_t)).uc_mcontext.gregs[REG_RIP as usize] += 3 }
```

Comment:
This causes the enclave execution not to be re-entered, but it does not kill the thread. But what happens in that worker thread in those cases?

Reply:
some garbage CoResult will be returned and sending it to IO queue will fail as sysloop is terminated by that moment

Reply:
I think you should adjust RAX and potentially other registers as well to something meaningful

Reply:
now enclave exiting flag is passed into coenter() to break enclu/AEX loop and special value CoResult::Abort will be returned if an enclave thread was interrupted that way. Why we may still need register adjustments?